modules/evm/src/lib.rs

```diff
@@ -467,21 +639,309 @@ impl<T: Trait> Module<T> {
+				if additional_storage > bytes {
+					Self::do_update_maintainer_storage_usage(
+						&contract_info.maintainer,
+						additional_storage,
+						additional_storage
+							.checked_add(bytes)
+							.expect("Non-negative integers sub can't overflow; qed"),
+					)?;
+				} else {
+					Self::do_update_maintainer_storage_usage(&contract_info.maintainer, additional_storage, 0)?;
+					account_info.storage_usage = Zero::zero();
+				}
+			}
+
+			Ok(())
+		})
+	}
+
+	fn do_remove_storage_quota(who: T::AccountId, contract: H160, bytes: u32) -> DispatchResult {
+		Accounts::<T>::mutate(contract, |maybe_account_info| -> DispatchResult {
+			let account_info = maybe_account_info.as_mut().ok_or(Error::<T>::ContractNotFound)?;
+			let contract_info = account_info
+				.contract_info
+				.as_ref()
+				.ok_or(Error::<T>::ContractNotFound)?;
+
+			let maintainer_account = T::AddressMapping::to_account(&contract_info.maintainer);
+			ensure!(who == maintainer_account, Error::<T>::NoPermission);
+
+			if bytes.is_zero() {
+				return Ok(());
+			}
+
+			let adjust_deposit = T::StorageDepositPerByte::get().saturating_mul(bytes.into());
+			ensure!(
+				account_info.storage_rent_deposit >= adjust_deposit,
+				Error::<T>::StorageQuotaNotEnough
+			);
+
+			account_info.storage_rent_deposit = account_info
+				.storage_rent_deposit
+				.checked_sub(&adjust_deposit)
+				.ok_or(Error::<T>::NumOutOfBound)?;
+			account_info.storage_quota = account_info
+				.storage_quota
+				.checked_sub(bytes)
+				.ok_or(Error::<T>::NumOutOfBound)?;
+
+			ensure!(
+				account_info.storage_usage <= account_info.storage_quota,
+				Error::<T>::StorageQuotaNotEnough
+			);
+
+			let additional_storage = {
+				let code_size = CodeInfos::get(contract_info.code_hash).map_or(0, |code_info| code_info.code_size);
+				contract_info
+					.total_storage_size()
+					.saturating_add(code_size)
+					.saturating_sub(account_info.storage_quota)
+			};
+
+			ensure!(additional_storage.is_zero(), Error::<T>::StorageQuotaNotEnough);
+			ensure!(
+				T::Currency::unreserve(&who, adjust_deposit).is_zero(),
+				Error::<T>::UnreserveFailed
+			);
+
+			Ok(())
+		})
+	}
+
+	fn do_request_transfer_maintainer(who: T::AccountId, contract: H160, new_maintainer: H160) -> DispatchResult {
+		Accounts::<T>::get(contract).map_or(Err(Error::<T>::ContractNotFound), |account_info| {
+			account_info
+				.contract_info
+				.map_or(Err(Error::<T>::ContractNotFound), |_| Ok(()))
+		})?;
+		ensure!(
+			PendingTransferMaintainers::<T>::get(contract, new_maintainer).is_none(),
+			Error::<T>::PendingTransferMaintainersExists
+		);
+
+		let transfer_maintainer_deposit = T::TransferMaintainerDeposit::get();
+		T::Currency::reserve(&who, transfer_maintainer_deposit)?;
+		PendingTransferMaintainers::<T>::insert(contract, new_maintainer, transfer_maintainer_deposit);
+		Ok(())
+	}
+
+	fn do_cancel_transfer_maintainer(who: T::AccountId, contract: H160, invalid_maintainer: H160) -> DispatchResult {
```

Comment:
```suggestion
	fn do_cancel_transfer_maintainer(who: T::AccountId, contract: H160, requester: H160) -> DispatchResult {
```
`invalid_maintainer` is a bit misleading